src/components/TxConfirmDialog/index.js

```diff
@@ -2,44 +2,104 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import _ from 'lodash';
+import { withStyles } from 'material-ui';
 import Button from 'material-ui/Button';
 import Dialog, { DialogTitle, DialogContent, DialogActions } from 'material-ui/Dialog';
 import { FormattedMessage, injectIntl, intlShape, defineMessages } from 'react-intl';
+import Table, {
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  TableFooter,
+} from 'material-ui/Table';
 
 import appActions from '../../redux/App/actions';
+import styles from './styles';
 
 const messages = defineMessages({
   confirmMessage: {
     id: 'txConfirm.message',
     defaultMessage: 'You are about to {txDesc} for {txAmount} {txToken}. Please click OK to continue.',
   },
+  confirmMessageWithFee: {
+    id: 'txConfirm.messageWithFee',
+    defaultMessage: 'You are about to {txDesc} for {txAmount} {txToken}, with fee for {txFee} QTUM. Please click OK to continue.',
+  },
 });
 
 @injectIntl
+@withStyles(styles, { withTheme: true })
 @connect((state) => ({
   txConfirmInfoAndCallback: state.App.get('txConfirmInfoAndCallback'),
+  transactionCost: state.App.get('transactionCost'),
 }), (dispatch) => ({
   clearTxConfirm: () => dispatch(appActions.clearTxConfirm()),
+  getTransactionCost: (txInfo) => dispatch(appActions.getTransactionCost(txInfo)),
 }))
+
```

Comment:
please keep these together.  The decorator is a function wrapping the component. We don't call functions like this do we?
```js
withStyles(styles, { withTheme: true })(
  TxConfirmDialog
);
```
no we do it like
```js
withStyles(styles, { withTheme: true })(TxConfirmDialog);
```
so let's keep them together please :)
```js
@withStyles(...)
export default ...
```